tensorflow/compiler/xla/service/gpu/cudnn_conv_rewriter.cc

```diff
@@ -248,7 +246,29 @@ std::tuple<bool, Window, ConvolutionDimensionNumbers> MatchBackwardFilter(
     backward_conv_dnums.add_kernel_spatial_dimensions(output_spatial_dims[i]);
   }
 
-  return std::make_tuple(true, backward_conv_window, backward_conv_dnums);
+  HloInstruction* lhs = conv->mutable_operand(0);
+  if (conv->feature_group_count() == 1) {
+    return std::make_tuple(true, backward_conv_window, backward_conv_dnums,
+                           lhs);
+  }
+  Shape new_shape = lhs->shape();
+
+  int64 input_batch_dimension = backward_conv_dnums.input_batch_dimension();
+  int64 input_feature_dimension = backward_conv_dnums.input_feature_dimension();
+
+  int64 input_batch = new_shape.dimensions(input_batch_dimension);
```

Comment:
Nit: You read and write new_shape which is a little confusing to read. What about the following:

```
int64 input_batch = lhs->shape().dimensions(input_batch_dimension);
int64 input_feature = lhs->shape().dimensions(input_feature_dimension);

Shape new_shape = lhs->shape();
new_shape.set_dimensions(input_batch_dimension,
￼                            input_batch / conv->feature_group_count());
```

Reply:
Done